<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LightWidget</class>
 <widget class="QWidget" name="LightWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>341</width>
    <height>490</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QListWidget" name="lightList"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="addLightButton">
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeLightButton">
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="lightTypeCombo">
        <item>
         <property name="text">
          <string>Directional Light</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Point Light</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Spot Light</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="labelPosition">
          <property name="text">
           <string>Position</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="positionX">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.989999999999782</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_11">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="positionY">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.989999999999782</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_12">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Z</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="positionZ">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="labelDirection">
          <property name="text">
           <string>Direction</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_13">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>X</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="directionX">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>5.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_14">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Y</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="directionY">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>5.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_15">
          <property name="maximumSize">
           <size>
            <width>15</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Z</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="directionZ">
          <property name="minimum">
           <double>-9999.000000000000000</double>
          </property>
          <property name="maximum">
           <double>9999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>5.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Diffuse Colour</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="diffusePushButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Specular Colour</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="SpecularPushButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Attenuation Range</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="rangeSpin">
          <property name="toolTip">
           <string>The radius of the light. i.e. A light bulb is a couple centimeters, the sun is ~696km;
            but be sure to be consistent in the unit of measurement you use (i.e. don't use km if
            your default unit is in meters).
            Note: Having a radius = 2 means that at distance = 2 the pixel is lit 100%
            Anything inside that radius (which is supposedly impossible) will have more than 100%
            light power.</string>
          </property>
          <property name="maximum">
           <double>99999.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Constant</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="constantSpin">
          <property name="toolTip">
           <string>Value in the range [0; 1)
            Sets range at which the luminance (in percentage) of a point would go below the threshold.
            For example lumThreshold = 0 means the attenuation range is infinity; lumThreshold = 1 means
            the range is set to 0.</string>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Attenuation Linear</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="linearSpin">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Quadratic</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="quadraticSpin">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Spotlight Inner Angle</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="innerAngleSpin">
          <property name="maximum">
           <double>180.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="outerAngleSpin">
          <property name="maximum">
           <double>180.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Outer Angle</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Falloff</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="falloffSpin">
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
